From a3768e9da7c0ce661252ffdc2cbfbfd134d3ef71 Mon Sep 17 00:00:00 2001
From: Pete Wyckoff <pw@osc.edu>
Date: Tue, 16 Oct 2007 15:21:33 -0400
Subject: [PATCH 5/7] local logging changes

Do not use syslog for error messages, just send them to stdout/stderr where
they can be redirected.  Add NDEBUG ifdef around log_debug to avoid
compiling those lines for optimized performance, and fix a compiler warning
that falls out due to this.  Put timestamps on messages for performance
debugging.
---
 usr/log.c |   19 +++++++++++++++----
 usr/log.h |    6 ++++++
 usr/spc.c |    2 ++
 3 files changed, 23 insertions(+), 4 deletions(-)

diff --git a/usr/log.c b/usr/log.c
index 414f9f4..f7cb240 100644
--- a/usr/log.c
+++ b/usr/log.c
@@ -24,9 +24,11 @@
 #include <unistd.h>
 #include <syslog.h>
 #include <signal.h>
+#include <time.h>
 #include <sys/shm.h>
 #include <sys/ipc.h>
 #include <sys/types.h>
+#include <sys/time.h>
 
 #include "log.h"
 
@@ -235,7 +237,7 @@ static void log_syslog (void * buff)
 
 static void dolog(int prio, const char *fmt, va_list ap)
 {
-	if (is_daemon) {
+	if (0 && is_daemon) {
 		la->ops[0].sem_op = -1;
 		if (semop(la->semid, la->ops, 1) < 0) {
 			syslog(LOG_ERR, "semop up failed");
@@ -250,7 +252,16 @@ static void dolog(int prio, const char *fmt, va_list ap)
 			return;
 		}
 	} else {
-		fprintf(stderr, "%s: ", log_name);
+		struct timeval tv;
+		time_t tp;
+		char buffer[16];
+
+		gettimeofday(&tv, NULL);
+		tp = tv.tv_sec;
+		strftime(buffer, 9, "%H:%M:%S", localtime(&tp));
+		sprintf(buffer+8, ".%06ld", (long) tv.tv_usec);
+
+		fprintf(stderr, "%s: [%s] ", log_name, buffer);
 		vfprintf(stderr, fmt, ap);
 		fflush(stderr);
 	}
@@ -308,7 +319,7 @@ int log_init(char *program_name, int size, int daemon, int debug)
 
 	logdbg(stderr,"enter log_init\n");
 	log_name = program_name;
-	if (is_daemon) {
+	if (0 && is_daemon) {
 		struct sigaction sa_old;
 		struct sigaction sa_new;
 		pid_t pid;
@@ -349,7 +360,7 @@ int log_init(char *program_name, int size, int daemon, int debug)
 
 void log_close (void)
 {
-	if (is_daemon) {
+	if (0 && is_daemon) {
 		closelog();
 		free_logarea();
 	}
diff --git a/usr/log.h b/usr/log.h
index 35ce374..40eb739 100644
--- a/usr/log.h
+++ b/usr/log.h
@@ -76,9 +76,15 @@ do {									\
 	log_error("%s(%d) " fmt, __FUNCTION__, __LINE__, ##args);	\
 } while (0)
 
+#ifndef NDEBUG
 #define dprintf(fmt, args...)						\
 do {									\
 	log_debug("%s(%d) " fmt, __FUNCTION__, __LINE__, ##args);	\
 } while (0)
+#else
+#define dprintf(fmt, args...)						\
+do {									\
+} while (0)
+#endif
 
 #endif	/* LOG_H */
diff --git a/usr/spc.c b/usr/spc.c
index 0d52164..8931b31 100644
--- a/usr/spc.c
+++ b/usr/spc.c
@@ -533,6 +533,7 @@ exit:
 
 void dump_cdb(struct scsi_cmd *cmd)
 {
+#ifndef NDEBUG
 	uint8_t *cdb = cmd->scb;
 
 	switch(cmd->scb_len) {
@@ -561,6 +562,7 @@ void dump_cdb(struct scsi_cmd *cmd)
 			cdb[12], cdb[13], cdb[14], cdb[15]);
 		break;
 	}
+#endif
 }
 
 int spc_illegal_op(int host_no, struct scsi_cmd *cmd)
-- 
1.5.3.6

